nomad/vault.go

```diff
@@ -1239,7 +1229,7 @@ func (v *vaultClient) revokeDaemon() {
 		case <-v.tomb.Dying():
 			return
 		case now := <-ticker.C:
-			if established, _ := v.ConnectionEstablished(); !established {
+			if established, err := v.ConnectionEstablished(); !established || err != nil {
```

Comment:
This code here was introduced in https://github.com/hashicorp/nomad/commit/42f7bc8e81ecb64d3aeb0f9f1936ff81059857dd#diff-87a4c76eca3de54716c55ba6b8c06475 - where `established` is true only when there is no error.  So checking error wasn't strictly necessary.  After many modifications, this file consistently checks for `established && err == nil`, but we missed this spot.

For some reason, not so clear to me, https://github.com/hashicorp/nomad/pull/3957 chose to represent invalid tokens as `established` with true but non-nil error.  It seems like we missed this spot.